### ISSUE TITLE
dartfmt should add braces if it splits a long if line

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/dart_style#664]
@devoncarew 
Tested on 2.0.0-dev.16.0.

If dartfmt sees a line like this:

```
    if (!identical(canonicalModelElement, this)) return canonicalModelElement?.href;
```

It will change it to:

```
   if (!identical(canonicalModelElement, this))
     return canonicalModelElement?.href;
```

I think it would be nice if this added braces.

**Comment by devoncarew** [comment]
+1, but I think @munificent generally prefers that dart_style only manipulate whitespace?

**Comment by munificent** [comment]
Yeah. I'm definitely sympathetic but, at least for now, it's considered out of scope. This has more detail:

https://github.com/dart-lang/dart_style/wiki/FAQ#why-doesnt-the-formatter-add-curlies-or-otherwise-clean-up-code

**Comment by natebosch** [comment]
I don't see a canonical issue discussing adding braces as a `--fix` so I'll reopen this one.

The scope of the formatter has increased since we added `--fix`. We should consider including this as one since it's tedious and we are opinionated about it.

**Comment by munificent** [comment]
Yeah, now that we have `--fix`, this is reasonable. Though it is somewhat tricky, for a couple of reasons:

* Should it be symmetric and *remove* curlies if an if does fit on one line? Some (most?) users might not want that. We could probably have two fixes, but that's a lot of effort. Note that `--fix` means "apply all the fixes", so if a significant number of users don't want it to remove curlies, we should probably just not have that as a fix at all.

* The fix requires knowing how a line gets split, but the formatter doesn't calculate line splits until after fixes have been applied, so this might be tricky to implement in a non-hacky way.

**Comment by natebosch** [comment]
> Should it be symmetric and _remove_ curlies if an if does fit on one line?

IMO no it shouldn't.

**Comment by munificent** [comment]
Closing this since we're removing `dart format --fix` in favor of `dart fix` (#1153). I believe the latter already supports this.
